packages/teleport/src/Nodes/AddNode/useAddNode.ts

```diff
@@ -16,41 +16,37 @@ limitations under the License.
 
 import { useState, useEffect } from 'react';
 import TeleportContext from 'teleport/teleportContext';
-import cfg from 'teleport/config';
 import useAttempt from 'shared/hooks/useAttemptNext';
 
-export default function useNodeAdd(ctx: TeleportContext, onClose: () => void) {
+export default function useAddNode(ctx: TeleportContext) {
   const { attempt, run } = useAttempt('processing');
   const canCreateToken = ctx.storeUser.getTokenAccess().create;
   const version = ctx.storeUser.state.cluster.authVersion;
-  const isEnterprise = cfg.isEnterprise;
-
+  const [automatic, setAutomatic] = useState(true);
   const [script, setScript] = useState('');
   const [expiry, setExpiry] = useState<Date>(null);
 
   useEffect(() => {
-    if (isEnterprise && canCreateToken) {
-      run(() => createJoinToken());
-    }
+    run(() => createJoinToken());
   }, []);
 
   function createJoinToken() {
-    return ctx.nodeService.createNodeJoinToken().then(token => {
-      setExpiry(token.expiry);
-      setScript(
-        `sudo bash -c "$(curl -fsSL ${cfg.getNodeJoinScriptUrl(token.id)})"`
-      );
+    return ctx.nodeService.createNodeBashCommand().then(cmd => {
+      setExpiry(cmd.expiry);
+      setScript(cmd.text);
```

Comment:
this function is exported, and the way i see it used, error isn't handled (sorry, this is my bug), probably means for the AddNode.Automatically.tsx:

```
if (attempt.status === 'failed') {
    return <Alert kind="danger" children={attempt.statusText} />;
  }
```
That has to be part of the main dialogue